src/iohumdrum.cpp

```diff
@@ -11591,7 +11594,7 @@ void HumdrumInput::addTextElement(
             }
         }
 
-        if (i < (int)pieces.size() - 1) {
+        if (i < piecesSize - 1) {
```

Comment:
This line of code is precisely why I do not use unsigned integers for indexes.  If `piecesSize` is `0`, then `piecesSize-1` is `MAX_INT`, and that is highly likely to be larger than `i`.  It does not matter if `piecesSize` is guaranteed to never be `0` at this case, because in the future something above the code could change, and maybe `0` would be allowed.  Hell will freeze over before I use `size_t` for small indexes in code where I am the primary maintainer (which is `iohumdrum.cpp`).  So: use type `int` for `piecesSize`.

Reply:
If `piecesSize` is 0 `for`-loop will not be executed. So we are safe.